ql/src/java/org/apache/hadoop/hive/ql/io/parquet/vector/VectorizedParquetRecordReader.java

```diff
@@ -278,12 +284,77 @@ private void checkEndOfRowGroup() throws IOException {
     }
     List<ColumnDescriptor> columns = requestedSchema.getColumns();
     List<Type> types = requestedSchema.getFields();
-    columnReaders = new VectorizedColumnReader[columns.size()];
-    for (int i = 0; i < columns.size(); ++i) {
-      columnReaders[i] =
-        new VectorizedColumnReader(columns.get(i), pages.getPageReader(columns.get(i)),
-          skipTimestampConversion, types.get(i));
+    columnReaders = new VectorizedParquetColumnReader[columns.size()];
+    for (int i = 0; i < columnTypesList.size(); ++i) {
+      columnReaders[i] = buildVectorizedParquetReader(columnTypesList.get(i), types.get(i), pages,
+        requestedSchema, skipTimestampConversion);
     }
     totalCountLoadedSoFar += pages.getRowCount();
   }
+
+  private List<ColumnDescriptor> getAllColumnDescriptorByType(
+    int depth,
+    Type type,
+    List<ColumnDescriptor> columns) {
+    List<ColumnDescriptor> res = new ArrayList<>();
+    for (ColumnDescriptor descriptor : columns) {
+      if (type.getName().equals(descriptor.getPath()[depth])) {
```

Comment:
It happens only when schema is corrupted. Addressing this by adding a check before this if block.